# Data Quality Assessment

**Project:** Google Play Store Business Analytics Dashboard

**Phase:** 02 - Data Quality Assessment

**Author:** Shruti T. Sakpal

**Objective:**
Evaluate the quality of the raw dataset by identifying missing values, duplicates, invalid entries, inconsistent data types, and other issues before performing any cleaning or analysis.

**Status:** In Progress

In [2]:
import pandas as pd 
import numpy as np

In [3]:
df = pd.read_csv("../data/raw/Google-Playstore.csv")

In [4]:
df.shape

(2312944, 24)

In [5]:
df.isnull().sum()

App Name                  5
App Id                    0
Category                  0
Rating                22883
Rating Count          22883
Installs                107
Minimum Installs        107
Maximum Installs          0
Free                      0
Price                     0
Currency                135
Size                    196
Minimum Android        6530
Developer Id             33
Developer Website    760835
Developer Email          31
Released              71053
Last Updated              0
Content Rating            0
Privacy Policy       420953
Ad Supported              0
In App Purchases          0
Editors Choice            0
Scraped Time              0
dtype: int64

In [6]:
missing_values = (
    df.isnull()
      .sum()
      .to_frame(name="Missing Count")
)

missing_values["Missing %"] = (
    missing_values["Missing Count"] / len(df) * 100
).round(2)

missing_values = missing_values.sort_values(
    by="Missing %",
    ascending=False
)

missing_values

,Missing Count,Missing %
Developer Website,760835,32.89
Privacy Policy,420953,18.20
Released,71053,3.07
Rating,22883,0.99
Rating Count,22883,0.99
Minimum Android,6530,0.28
Currency,135,0.01
Size,196,0.01
Maximum Installs,0,0.00
Minimum Installs,107,0.00


## Key Findings

- Developer Website has the highest missing percentage (32.89%).
- Privacy Policy is missing for 18.20% of apps.
- Rating-related fields have less than 1% missing data.
- Core business attributes are largely complete.

### Business Impact

The missing values are concentrated in optional metadata rather than core analytical fields, suggesting the dataset is suitable for marketplace and performance analysis after targeted cleaning.

In [7]:
duplicate_rows = df.duplicated().sum()

print(f"Number of duplicate rows: {duplicate_rows}")

Number of duplicate rows: 0


In [8]:
duplicate_app_ids = df["App Id"].duplicated().sum()

print(f"Number of duplicate App IDs: {duplicate_app_ids}")

Number of duplicate App IDs: 0


## Key Findings

- ✅ No duplicate rows found.
- ✅ No duplicate App IDs found.
- ⚠️ Some columns have incorrect data types and require conversion during the cleaning phase.

### Business Impact

The dataset has strong structural integrity, reducing the risk of duplicate-driven bias. The primary data quality concern now shifts toward datatype standardization, which is essential for accurate time-series and numerical analysis.

In [9]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2312944 entries, 0 to 2312943
Data columns (total 24 columns):
 #   Column             Dtype  
---  ------             -----  
 0   App Name           str    
 1   App Id             str    
 2   Category           str    
 3   Rating             float64
 4   Rating Count       float64
 5   Installs           str    
 6   Minimum Installs   float64
 7   Maximum Installs   int64  
 8   Free               bool   
 9   Price              float64
 10  Currency           str    
 11  Size               str    
 12  Minimum Android    str    
 13  Developer Id       str    
 14  Developer Website  str    
 15  Developer Email    str    
 16  Released           str    
 17  Last Updated       str    
 18  Content Rating     str    
 19  Privacy Policy     str    
 20  Ad Supported       bool   
 21  In App Purchases   bool   
 22  Editors Choice     bool   
 23  Scraped Time       str    
dtypes: bool(4), float64(4), int64(1), str(15)
memory usage: 361.8

## Key Findings

- Most columns already have appropriate data types.
- Five columns require datatype conversion before analysis:
  - Installs
  - Rating Count
  - Minimum Installs
  - Released
  - Last Updated
  - Scraped Time
- The Size column requires additional preprocessing because it contains values with units (e.g., MB, KB) rather than pure numeric values.

### Business Impact

Correct data types are essential for accurate numerical calculations and time-based analysis. These conversions will be performed during the Data Cleaning phase after all quality issues have been assessed.